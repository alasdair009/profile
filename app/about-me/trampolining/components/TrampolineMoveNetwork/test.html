<html lang="en">
  <head>
    <style>
      .videos {
        display: flex;
        height: 100vh;
        width: 100vw;
      }
      .videos div {
        align-items: center;
        display: flex;
        flex-direction: column;
        justify-content: center;
        width: 50%;
      }
      iframe {
        aspect-ratio: 16/9;
        width: 90%;
      }
    </style>
  </head>
  <body>
    <div class="videos">
      <div>
        <iframe
          id="standard"
          src=""
          data-src="https://www.youtube.com/embed/NpEaa2P7qZI?si=uJjK4C2TLWivF5BX"
          title="Standard Embed"
          allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture; web-share"
          referrerpolicy="strict-origin-when-cross-origin"
          allowfullscreen
        ></iframe>
        <button id="goStandard">Standard</button>
      </div>
      <div>
        <iframe
          id="privacy"
          src=""
          data-src="https://www.youtube-nocookie.com/embed/NpEaa2P7qZI?si=uJjK4C2TLWivF5BX"
          title="Enhanced privacy"
          allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture; web-share"
          referrerpolicy="strict-origin-when-cross-origin"
          allowfullscreen
        ></iframe>
        <button id="goPrivacy">Privacy</button>
      </div>
    </div>
    <script>
      const goStandard = document.getElementById("goStandard");
      const standardIFrame = document.getElementById("standard");
      const goPrivacy = document.getElementById("goPrivacy");
      const privacyIFrame = document.getElementById("privacy");
      goStandard.addEventListener(
        "click",
        function () {
          standardIFrame.setAttribute("src", standardIFrame.getAttribute("data-src"));
        },
        false
      );
      goPrivacy.addEventListener(
        "click",
        function () {
          privacyIFrame.setAttribute("src", privacyIFrame.getAttribute("data-src"));
        },
        false
      );
    </script>
  </body>
</html>
