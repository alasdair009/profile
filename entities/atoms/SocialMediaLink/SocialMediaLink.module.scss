@use "../../../app/styles/animation";
@use "../../../app/styles/colors";
@use "../../../app/styles/dimensions";
@use "../../../app/styles/helpers/rem";
@use "sass:map";

$socialMediaIcons: (
  bluesky: url(./assets/bluesky.svg),
  facebook: url(./assets/facebook.svg),
  github: url(./assets/github.svg),
  instagram: url(./assets/instagram.svg),
  linkedin: url(./assets/linkedin.svg),
  x: url(./assets/x.svg),
  youtube: url(./assets/youtube.svg),
);

.root {
  align-items: center;
  aspect-ratio: 1;
  background: colors.$colorblackEvil;
  color: transparent;
  display: flex;
  justify-content: center;
  -webkit-mask: no-repeat center;
  mask: no-repeat center;
  -webkit-mask-size: contain;
  mask-size: contain;
  overflow: hidden;
  position: relative;
  width: rem.toRem(map.get(dimensions.$sizes, s48));

  &::before {
    background: linear-gradient(
        colors.$colorWhiteGhost 50%,
        colors.$colorGreenGrass 50%,
        colors.$colorGreenGrass 100%
      )
      0 0;
    content: "";
    height: 200%;
    left: 0;
    position: absolute;
    top: 0;
    transition: all animation.animationDuration(0.25);
    width: 100%;
  }

  &:hover::before {
    top: -100%;
  }
}
