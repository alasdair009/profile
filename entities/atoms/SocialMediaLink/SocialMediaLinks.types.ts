export type SocialMediaLinkVariant = "linkedin" | "x";
