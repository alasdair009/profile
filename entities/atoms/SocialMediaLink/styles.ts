"use client";
import styled from "styled-components";
import Link from "next/link";
import { SocialMediaLinkVariant } from "./SocialMediaLinks.types";
import { animationDurationCSS, colors, sizes } from "@/entities";
import xIcon from "./assets/x.svg";
import instagramIcon from "./assets/instagram.svg";
import linkedInIcon from "./assets/linkedin.svg";
import gitHubIcon from "./assets/github.svg";
import facebookIcon from "./assets/facebook.svg";
import youTubeIcon from "./assets/youtube.svg";
import blueskyIcon from "./assets/bluesky.svg";

const getMaskIcon = (variant: SocialMediaLinkVariant): string => {
  switch (variant) {
    case "bluesky":
      return blueskyIcon.src;
    case "facebook":
      return facebookIcon.src;
    case "github":
      return gitHubIcon.src;
    case "instagram":
      return instagramIcon.src;
    case "linkedin":
      return linkedInIcon.src;
    case "x":
      return xIcon.src;
    case "youtube":
      return youTubeIcon.src;
  }
};

export const Root = styled(Link)<{ $variant: SocialMediaLinkVariant }>`
  align-items: center;
  aspect-ratio: 1;
  background: black;
  color: transparent;
  display: flex;
  justify-content: center;
  -webkit-mask: url(${({ $variant }) => getMaskIcon($variant)}) no-repeat center;
  mask: url(${({ $variant }) => getMaskIcon($variant)}) no-repeat center;
  -webkit-mask-size: contain;
  mask-size: contain;
  overflow: hidden;
  position: relative;
  width: ${sizes.s48.rem};
  
  &::before {
    background: linear-gradient(${colors.whiteGhost} 50%, ${colors.greenGrass} 50%, ${colors.greenGrass} 100%) 0 0;
    content: '';
    height: 200%;
    left: 0;
    position: absolute;
    top: 0;
    transition: all ${animationDurationCSS(0.25)};
    width: 100%;
  }

  &:hover::before {
    top: -100%;
  }
}
`;
