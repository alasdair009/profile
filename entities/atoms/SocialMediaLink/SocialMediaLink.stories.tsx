import { SocialMediaLink } from "./SocialMediaLink";
import { Meta, StoryObj } from "@storybook/react";
import { expect, userEvent, within } from "@storybook/test";

const meta: Meta<typeof SocialMediaLink> = {
  component: SocialMediaLink,
};
export default meta;

export const X: StoryObj<typeof SocialMediaLink> = {
  args: {
    variant: "x",
  },
  play: async ({ canvasElement }) => {
    const canvas = within(canvasElement);
    const socialMediaElement = canvas.getByTestId(SocialMediaLink.name);

    await expect(socialMediaElement).toBeInTheDocument();
    await expect(socialMediaElement).toHaveAttribute(
      "href",
      "https://twitter.com/alasdair009"
    );
  },
};

export const LinkedIn: StoryObj<typeof SocialMediaLink> = {
  args: {
    variant: "linkedin",
  },
  play: async ({ args, canvasElement }) => {
    const canvas = within(canvasElement);
    const socialMediaElement = canvas.getByTestId(SocialMediaLink.name);

    await expect(socialMediaElement).toBeInTheDocument();
    await expect(socialMediaElement).toHaveAttribute(
      "href",
      "https://www.linkedin.com/in/alasdairmacrae/"
    );
  },
};

export const GitHub: StoryObj<typeof SocialMediaLink> = {
  args: {
    variant: "github",
  },
  play: async ({ args, canvasElement }) => {
    const canvas = within(canvasElement);
    const socialMediaElement = canvas.getByTestId(SocialMediaLink.name);

    await expect(socialMediaElement).toBeInTheDocument();
    await expect(socialMediaElement).toHaveAttribute(
      "href",
      "https://github.com/alasdair009"
    );
  },
};

export const Facebook: StoryObj<typeof SocialMediaLink> = {
  args: {
    variant: "facebook",
  },
  play: async ({ args, canvasElement }) => {
    const canvas = within(canvasElement);
    const socialMediaElement = canvas.getByTestId(SocialMediaLink.name);

    await expect(socialMediaElement).toBeInTheDocument();
    await expect(socialMediaElement).toHaveAttribute(
      "href",
      "https://www.facebook.com/alasdair009"
    );
  },
};

export const Instagram: StoryObj<typeof SocialMediaLink> = {
  args: {
    variant: "instagram",
  },
  play: async ({ args, canvasElement }) => {
    const canvas = within(canvasElement);
    const socialMediaElement = canvas.getByTestId(SocialMediaLink.name);

    await expect(socialMediaElement).toBeInTheDocument();
    await expect(socialMediaElement).toHaveAttribute(
      "href",
      "https://www.instagram.com/alasdair009"
    );
  },
};

export const YouTube: StoryObj<typeof SocialMediaLink> = {
  args: {
    variant: "youtube",
  },
  play: async ({ args, canvasElement }) => {
    const canvas = within(canvasElement);
    const socialMediaElement = canvas.getByTestId(SocialMediaLink.name);

    await expect(socialMediaElement).toBeInTheDocument();
    await expect(socialMediaElement).toHaveAttribute(
      "href",
      "https://www.youtube.com/alasdair009"
    );
  },
};
