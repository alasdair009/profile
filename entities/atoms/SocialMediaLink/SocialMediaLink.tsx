import { Root } from "./styles";
import { HTMLAttributes } from "react";
import { SocialMediaLinkVariant } from "./SocialMediaLinks.types";

type SocialMediaProps = {
  /**
   * Network to link to.
   */
  variant: SocialMediaLinkVariant;
} & HTMLAttributes<HTMLAnchorElement>;

const getSocialMediaUrl = (variant: SocialMediaLinkVariant) => {
  switch (variant) {
    case "github":
      return "https://github.com/alasdair009";
    case "linkedin":
      return "https://www.linkedin.com/in/alasdairmacrae/";
    case "x":
      return "https://twitter.com/alasdair009";
  }
};

/**
 * Icon that links to the requested social media network.
 */
export function SocialMediaLink({ variant, children, ...rest }: SocialMediaProps) {
  return (
    <Root href={getSocialMediaUrl(variant)} target="_blank" rel="noopener" $variant={variant} {...rest}>
      {children}
    </Root>
  );
}
