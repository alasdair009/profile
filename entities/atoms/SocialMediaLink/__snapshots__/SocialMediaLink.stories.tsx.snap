// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`atoms/SocialMediaLink GitHub play-test 1`] = `
<a target="_blank"
   rel="noopener"
   data-testid="SocialMediaLink"
   class="sc-hDcvty kFHYFj"
   href="https://github.com/alasdair009"
>
</a>
`;

exports[`atoms/SocialMediaLink LinkedIn play-test 1`] = `
<a target="_blank"
   rel="noopener"
   data-testid="SocialMediaLink"
   class="sc-hDcvty hrIVTH"
   href="https://www.linkedin.com/in/alasdairmacrae/"
>
</a>
`;

exports[`atoms/SocialMediaLink X play-test 1`] = `
<a target="_blank"
   rel="noopener"
   data-testid="SocialMediaLink"
   class="sc-hDcvty kQMZGL"
   href="https://twitter.com/alasdair009"
>
</a>
`;
